config/notifiers.go

```diff
@@ -625,3 +634,39 @@ func (c *SNSConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	}
 	return nil
 }
+
+// TelegramConfig configures notifications via Telegram.
+type TelegramConfig struct {
+	NotifierConfig `yaml:",inline" json:",inline"`
+
+	HTTPConfig *commoncfg.HTTPClientConfig `yaml:"http_config,omitempty" json:"http_config,omitempty"`
+
+	APIUrl               *URL   `yaml:"api_url" json:"api_url,omitempty"`
+	BotToken             string `yaml:"bot_token,omitempty" json:"token,omitempty"`
+	ChatID               int64  `yaml:"chat_id,omitempty" json:"chat,omitempty"`
+	Message              string `yaml:"message,omitempty" json:"message,omitempty"`
+	DisableNotifications bool   `yaml:"disable_notifications,omitempty" json:"disable_notifications,omitempty"`
+	ParseMode            string `yaml:"parse_mode,omitempty" json:"parse_mode,omitempty"`
+}
+
+// UnmarshalYAML implements the yaml.Unmarshaler interface.
+func (c *TelegramConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	*c = DefaultTelegramConfig
+	type plain TelegramConfig
+	if err := unmarshal((*plain)(c)); err != nil {
+		return err
+	}
+	if c.BotToken == "" {
+		return fmt.Errorf("missing bot_token on telegram_config")
+	}
+	if c.ChatID == 0 {
+		return fmt.Errorf("missing chat_id on telegram_config")
+	}
+	if c.ParseMode != "" &&
+		c.ParseMode != "Markdown" &&
+		c.ParseMode != "MarkdownV2" &&
+		c.ParseMode != "HTML" {
+		return fmt.Errorf("unknown parse_mode on telegram_config")
```

Comment:
Would it be too verbose to list the valid modes in the error?

Reply:
Yes, you're right, added all possible values to the log message